lib/bundlex.ex

```diff
@@ -8,6 +8,27 @@ defmodule Bundlex do
 
   @type platform_t :: :linux | :macosx | :windows32 | :windows64
 
+  @typedoc """
+  A tuple of three elements describing the platform.
+
+  It consists of:
+  * architecture - e.g. `x86_64` or `arm64`
+  * vendor - e.g. `pc`
+  * operating system - e.g. `linux` or `freebsd`
+  """
+  @type target_triplet ::
+          {architecture :: String.t(), vendor :: String.t(), operating_system :: String.t()}
```

Comment:
I'd make it a struct for more flexibility and easier access to fields. Also, maybe we could put the version of the operating system in a separate field, so we didn't have to pattern match it. I'd consider converting strings to atoms as well.

Reply:
That is also why I wouldn't convert the names into atoms, as it will make such a pattern matching more difficult (or impossible)

Reply:
Hmm, so maybe we could have a function like `Bundlex.family()` that would convert the OS to one of the well-known values? Then we could gradually improve it to support more systems if needed